src/components/contextual/pages/pool/PoolChart.vue

```diff
@@ -1,180 +1,209 @@
+  loading: false
+});
 
-export default defineComponent({
-  name: 'PoolChart',
-
-  props: {
-    prices: {
-      type: Object as PropType<Record<string, number[]>>,
-      required: true
-    },
-    snapshots: {
-      type: Object as PropType<PoolSnapshots>,
-      required: true
-    },
-    loading: { type: Boolean, default: true }
-  },
-
-  setup(props) {
-    const store = useStore();
-    const { t } = useI18n();
-
-    const {
-      prices,
-      snapshots
-    }: {
-      prices: Ref<Record<string, number[]>>;
-      snapshots: Ref<PoolSnapshots>;
-    } = toRefs(props);
-
-    const appLoading = computed(() => store.state.app.loading);
-    const tailwind = useTailwind();
-    const { darkMode } = useDarkMode();
-
-    const hodlColor = computed(() =>
-      darkMode.value
-        ? tailwind.theme.colors.gray['600']
-        : tailwind.theme.colors.black
-    );
-
-    const chartColors = [tailwind.theme.colors.green['400'], hodlColor.value];
-
-    const nonEmptySnapshots = computed(() => {
-      if (!history.value || !history.value) return [];
-      return history.value.filter((state: any) => state.totalShares !== '0');
-    });
+/**
+ * STATE
+ */
+const { prices, snapshots, pool } = toRefs(props);
+const MIN_CHART_VALUES = 7;
+
+/**
+ * COMPOSABLES
+ */
+const store = useStore();
+const { t } = useI18n();
+const tailwind = useTailwind();
+const { darkMode } = useDarkMode();
+
+/**
+ * COMPUTED
+ */
+const hodlColor = computed(() =>
+  darkMode.value
+    ? tailwind.theme.colors.gray['600']
+    : tailwind.theme.colors.black
+);
+
+const chartColors = computed(() => [
+  tailwind.theme.colors.green['400'],
+  hodlColor.value
+]);
+
+const supportsPoolLiquidity = computed(() =>
+  isStablePhantom(pool.value.poolType)
+);
+
+const appLoading = computed(() => store.state.app.loading);
+
+const history = computed(() => {
+  const pricesTimestamps = Object.keys(prices.value);
+  const snapshotsTimestamps = Object.keys(snapshots.value);
+
+  if (snapshotsTimestamps.length === 0) {
+    return [];
+  }
+
+  // Prices are not required when using pool liquidity
+  if (!supportsPoolLiquidity.value && pricesTimestamps.length === 0) {
+    return [];
+  }
+
+  return snapshotsTimestamps
+    .map(snapshotTimestamp => {
+      const timestamp = parseInt(snapshotTimestamp);
+
+      const snapshot = snapshots.value[timestamp];
+      const price = prices.value[timestamp] ?? [0, 0];
+      const amounts = snapshot.amounts ?? ['0', '0'];
+      const totalShares = parseFloat(snapshot.totalShares) ?? 0;
+      const liquidity = parseFloat(snapshot.liquidity) ?? 0;
+
+      return {
+        timestamp,
+        price,
+        amounts,
+        totalShares,
+        liquidity
+      };
+    })
+    .filter(state => state.totalShares > 0);
+});
+
+const timestamps = computed(() => {
+  if (history.value.length === 0) {
+    return [];
+  }
+
+  return history.value.map(state =>
+    format(fromUnixTime(state.timestamp / 1000), 'yyyy/MM/dd')
```

Comment:
Interesting that you still need to divide by 1000 in a fromUnixTime fn. I started adding helper functions in `useTime` maybe you could add something more universal to handle this case there?

Reply:
Yeah I don't think its needed actually because the timestamp is already normalized... so we can just `format` it.

BTW, unrelated by we seem to also have a `date.ts` which does a similar thing... we should merge them.